pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -75,6 +83,88 @@ let
             ++ lib.optional (config.pulseaudio or true) libpulseaudio;
       gtk_modules = [ libcanberra-gtk2 ];
 
+      #########################
+      #                       #
+      #   EXTRA PREF CHANGES  #
+      #                       #
+      #########################
+      policiesJson = builtins.toFile "policies.json"
+        (builtins.toJSON enterprisePolicies);
+
+      extensions = builtins.map (a:
+        if ! (builtins.hasAttr "name" a) || ! (builtins.isString a.name) then
+          throw "Firefox addon needs a name attribute"
+        else if ! (builtins.hasAttr "url" a) || ! (builtins.isString a.url) then
+          throw "Addon ${a.pname} needs an url"
+        else if ! (builtins.hasAttr "sha256" a) || ! (builtins.isString a.sha256) then
+          throw "Addon ${a.pname} needs an sha256 checksum"
+        else stdenv.mkDerivation rec {
+          pname = a.name;
+          version = "1.0";
+          src = fetchurl {
+            url = a.url;
+            sha256 = a.sha256;
+          };
+
+          phases = [ "buildPhase" ];
+
+          extid = "${a.sha256}@${a.name}";
+
+          buildInputs = [ zip unzip jq ];
+
+          buildPhase = ''
+            UUID="${extid}"
```

Comment:
Ok. How about using `UUID=$(sha256sum $src)` to get the UUID?

Reply:
We can't generate it dynamically because the name has to be statically present to whitelist it in line 142